### PR TITLE
Fix error where querySourceFeatures has parameters omitted. #3540

### PR DESCRIPTION
**Pull request by andrewharvey**
## Launch Checklist

 - [x] briefly describe the changes in this PR

See #3540 for background on what this fixes.

There are a couple of places where this could be fixed, so I'm not sure where is best higher or lower in the call stack. I've opted to place it at the bottom of the stack.

 - [x] write tests for all new functionality

The test is quite verbose because if you don't add any features to the source and if this source isn't included in a source the code path which I ran into this error isn't triggered.

If you remove the fix the test fails if you include this fix the test passes.

 - [x] document any changes to public APIs

I've included a second commit which makes `parameters` optional although I'm not sure if this is correct as it's only optional for GeoJSON sources as per current documentation.

I needed to rename `params` as otherwise it would show up in the generated docs separately.

 - [ ] post benchmark scores
 - [ ] manually test the debug page

### CODE REVIEW

**Comment by mourner** [comment]
Can you please fix the linting errors?

**Comment by andrewharvey** [comment]
> Can you please fix the linting errors?

Sorry I ran the unit tests but failed to run the full test suite.

Should be fixed now, I just overwrote the commits so not sure if ci automatically reruns or not...

File: js/ui/map.js [hunk 1]
```diff
@@ -575,7 +575,7 @@ class Map extends Camera {
      * representing features within the specified vector tile or GeoJSON source that satisfy the query parameters.
      *
      * @param {string} sourceID The ID of the vector tile or GeoJSON source to query.
-     * @param {Object} parameters
+     * @param {Object} [parameters]
```
Comment (lucaswoj) [inline js/ui/map.js]: Got it. This is more of a feature addition than a bug fix. We never supported or claimed to support omitting `parameters`. 👍

File: test/js/ui/map.test.js [hunk 2]
```diff
@@ -646,6 +647,50 @@ test('Map', (t) => {
         t.end();
     });
 
+    t.test('#querySourceFeatures', (t) => {
```
Comment (lucaswoj) [inline test/js/ui/map.test.js]: `querySourceFeatures` tests are currently split between [`tile.test.js`](http://github.com/mapbox/mapbox-gl-js/blob/0088929dcc934d1f41d5aec5c5bfe9bdd165f89f/test/js/source/tile.test.js#L38-L39) and [`style.test.js`](http://github.com/mapbox/mapbox-gl-js/blob/0088929dcc934d1f41d5aec5c5bfe9bdd165f89f/test/js/style/style.test.js#L1179-L1180). Could you move this test to one of those two places?

File: js/ui/map.js (hunk 1 above)
Comment (andrewharvey) [inline js/ui/map.js]: I suppose so it is :wink: 

Given `parameters.sourceLayer` and `parameters.filter` were optional I incorrectly assumed I could omit `parameters` entirely but I guess the JS doc says you still need to provide an empty object.

**Comment by lucaswoj** [comment]
Looks great! Thank you so much!

**Pull request merged**